src/context/MapContext.tsx

```diff
@@ -84,6 +98,60 @@ export const MapProvider = ({ initialState = {}, children }: MapProviderProps) =
     [activeFilters],
   );
 
+  useEffect(() => {
+    if (!map) return;
+    const source = new VectorSource();
+
+    const layer = new VectorLayer({
+      source: source,
+      style: {
+        'fill-color': 'rgba(255, 255, 255, 0.2)',
+        'stroke-color': '#ffcc33',
+        'stroke-width': 2,
+        'circle-radius': 7,
+        'circle-fill-color': '#ffcc33',
+      },
+    });
+    layer.set('name', 'aoiLayer');
+    layer.setZIndex(10);
+    layer.setVisible(true);
+
+    map.addLayer(layer);
+    vectorLayer.current = layer;
+
+    if (drawingSource) return;
+    setDrawingSource(source);
+  }, [map, drawingSource]);
+
+  useEffect(() => {
+    if (!map) return;
+    setQueryParam('view', 'map');
+    map.on('moveend', () => {
+      const view = map.getView();
+      const centre = view.getCenter();
+      setQueryParam('centre', centre.toString());
+    });
+    map.getView().on('change:resolution', () => {
+      const zoom = map.getView().getZoom();
+      setQueryParam('zoom', zoom.toString());
+    });
+  }, [map]);
+
+  useEffect(() => {
+    if (!map) return;
+    if (initialMapValues) return;
+    const mapView = map.getView();
+    const centre = getQueryParam('centre');
+    if (centre) {
+      mapView.setCenter([parseFloat(centre.split(',')[0]), parseFloat(centre.split(',')[1])]);
```

Comment:
For future, we should have a dedicated place for map utilities like this as theres a lot of map logic spread about